src/main/java/ekud/task/Task.java

```diff
@@ -0,0 +1,175 @@
+package ekud.task;
+
+import java.util.HashMap;
+
+import ekud.components.TaskList;
+import ekud.exceptions.EkudException;
+import ekud.ui.Ui;
+
+/**
+ * Represents a task. Each task minimally has a description and completion status.
```

Comment:
Consider adding to the JavaDoc that Task is an abstract class and not meant to be instantiated.